### PR TITLE
gh-98739: Update libexpat from 2.4.9 to 2.5.0

### PR DESCRIPTION
**Pull request by scdub**
Based this PR on the recent earlier upgrade at https://github.com/python/cpython/pull/97006 by @corona10 

* Issue: gh-98739

### CODE REVIEW

**Comment by scdub** [comment]
@hartwork I'm not sure if there are other changes that you think would be necessary for Python to upgrade to your latest release, but here's a first attempt at to get Python synced up. I believe there is a constellation of security fixes which Python will be requiring in the near term (SQLite, OpenSSL, ...) and this could potentially tag along for one of those releases.

**Comment by hartwork** [comment]
@scdub I don't really understand your message. Can you maybe rephrase your question for me to better understand?

**Review by hartwork — APPROVED** [review]
Verified using this Dockerfile:
```Dockerfile
# Copyright (c) 2022 Sebastian Pipping <sebastian@pipping.org>
# Licensed under the Apache License version 2.0

FROM alpine
RUN apk add --update \
            diffutils \
            git \
            sed \
        && \
    git clone --depth 1 https://github.com/python/cpython cpython-main \
        && \
    ( cd cpython-main && git rev-parse HEAD ) \
        && \
    git clone --depth 1 --branch expat-2.5.0 https://github.com/scdub/cpython scdub-expat-2-5-0 \
        && \
    ( cd scdub-expat-2-5-0 && git rev-parse HEAD ) \
        && \
    git config --global advice.detachedHead false \
        && \
    git clone --depth 1 --branch R_2_4_9 https://github.com/libexpat/libexpat libexpat_2_4_9 \
        && \
    git clone --depth 1 --branch R_2_5_0 https://github.com/libexpat/libexpat libexpat_2_5_0 \
        && \
    diff -r -u libexpat_2_4_9/expat/lib/ cpython-main/Modules/expat/ | tee 2-4-9.diff \
        && \
    diff -r -u libexpat_2_5_0/expat/lib/ scdub-expat-2-5-0/Modules/expat/ | tee 2-5-0.diff \
        && \
    sed -e '/^Only in /d' -e '/^\(+++\|---\) /d' -e '/^diff /d' -i 2-4-9.diff 2-5-0.diff \
        && \
    diff -u 2-4-9.diff 2-5-0.diff \
        && \
    echo 'Diff is good.'
```

**Comment by scdub** [comment]
> @scdub I don't really understand your message. Can you maybe rephrase your question for me to better understand?

I applogize, I wasn't clear. I noticed that with 2.5.0, there is a new `EXPAT_CONFIG_H` definition for preventing multiple inclusions and wasn't sure if there was any changes required for [`pyexpatns.h`](https://github.com/python/cpython/blob/3db91f10cccc09949a96f3935540a0938d0b4de8/Modules/expat/pyexpatns.h) or they can remain independent approaches.

**Comment by hartwork** [comment]
@scdub thanks for elaborating. The added `EXPAT_CONFIG_H` protection originally came up in relation to refactoring the test suite that triggered compiler warnings on redefining existing macros. I think it can safely be ignored in context of CPython.

**Comment by scdub** [comment]
@gpshead I'm not sure if this PR is of interest to you, but saw that you did initiate the work for the previous expat upgrade to 2.4.9.

**Comment by gpshead** [comment]
Confirmed that these match the libexpat 2.5.0 upstream sources from `SHA256(expat-2.5.0.tar.xz)= ef2420f0232c087801abf705e89ae65f6257df6b7931d37846a193ef2e8cdcbe`

with our pyexpatns.h addition and that this adds no new C APIs that need including in that "namespace" header hack.

**Review by gpshead — APPROVED** [review]
